daemon/daemon.go

```diff
@@ -19,6 +19,101 @@
 
 package daemon
 
+/*
+#cgo pkg-config: libsystemd
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <systemd/sd-bus.h>
+
+// Yes, this is as hacky as it looks. This copies the same method from libpolkit
+static uint64_t
+get_start_time (pid_t pid)
+{
+    char filename[1024], line[1024];
+    FILE *f = NULL;
+    int i, n_spaces;
+    uint64_t start_time = 0;
+
+    snprintf (filename, 1024, "/proc/%u/stat", pid);
+    f = fopen (filename, "r");
+    if (!fgets (line, 1024, f))
+        goto done;
+
+    // A line looks like this (the 22nd field is the start time - 545930 in this case):
+    // 14207 (gnome-terminal-) S 3399 3741 3741 0 -1 4194304 95756 7917395 23 857 6850 942 29913 2486 20 0 4 0 545930 692899840 12642 18446744073709551615 4194304 4486204 140735647896976 140735647896432 139643272960061 0 0 4096 65536 0 0 0 17 3 0 0 1 0 0 6585664 6596840 22155264 140735647901892 140735647901938 140735647901938 140735647903690 0
+
```

Comment:
strrchr? Yes. I think initially I was using that and it had too pointer arithmetic but it looks better now.